docs/source/developers/crossbow.rst

```diff
@@ -0,0 +1,284 @@
+   ``crossbow``):
+
+   -  TravisCI: ``https://travis-ci.org/<ghuser>/<ghrepo>/settings``
+   -  Appveyor:
+      ``https://ci.appveyor.com/project/<ghuser>/<ghrepo>/settings/environment``
+   -  CircleCI:
+      ``https://circleci.com/gh/<ghuser>/<ghrepo>/edit#env-vars``
+
+   On Appveyor check the ``skip branches without appveyor.yml`` checkbox
+   on the web UI under crossbow repository’s settings.
+
+7. Install Python 3.6:
+
+   Miniconda is preferred, see installation instructions:
+   https://conda.io/docs/user-guide/install/index.html
+
+8. Install the python dependencies for the script:
+
+   .. code:: bash
+
+      conda install -c conda-forge -y \
+          jinja2 \
+          pygit2 \
+          click \
+          ruamel.yaml \
+          setuptools_scm \
+          github3.py \
+          python-gnupg \
+          toolz \
+          jira
+
+   .. code:: bash
+
+      # pygit2 requires libgit2: http://www.pygit2.org/install.html
+      pip install \
+          jinja2 \
+          pygit2 \
+          click \
+          ruamel.yaml \
+          setuptools_scm \
+          github3.py \
+          python-gnupg \
+          toolz \
+          jira
+
+9. Try running it:
+
+   .. code:: bash
+
+      $ python crossbow.py --help
+
+Usage
+-----
+
+The script does the following:
+
+1. Detects the current repository, thus supports forks. The following
+   snippet will build kszucs’s fork instead of the upstream apache/arrow
+   repository.
+
+   .. code:: bash
+
+      $ git clone https://github.com/kszucs/arrow
+      $ git clone https://github.com/kszucs/crossbow
+
+      $ cd arrow/dev/tasks
+      $ python crossbow.py submit conda-win conda-linux conda-osx
+
+2. Gets the HEAD commit of the currently checked out branch and
+   generates the version number based on `setuptools_scm`_. So to build
+   a particular branch, just check out before running the script:
+
+   .. code:: bash
+
+      git checkout ARROW-<ticket number>
+      python dev/tasks/crossbow.py submit --dry-run conda-linux conda-osx
+
+   ..
+
+      Note that the arrow branch must be pushed beforehand, because the
+      script will clone the selected branch.
+
+3. Reads and renders the required build configurations with the
+   parameters substituted.
+
+4. Create a branch per task, prefixed with the job id. For example to
+   build conda recipes on linux it will create a new branch:
+   ``crossbow@build-<id>-conda-linux``.
+
+5. Pushes the modified branches to GitHub which triggers the builds. For
+   authentication it uses GitHub OAuth tokens described in the install
+   section.
+
+Query the build status
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   python crossbow.py status <build id / branch name>
+
+Download the build artifacts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   python crossbow.py artifacts <build id / branch name>
+
+Examples
+~~~~~~~~
+
+The script accepts a pattern as a first argument to narrow the build
```

Comment:
Looks like this is missing its example. No idea what "pattern" or "first argument" means here.